src/modules/job-exec/job-exec.h

```diff
@@ -46,17 +46,23 @@ struct jobinfo;
  *
  *   - stats:   (optional) get json object of exec implementation stats
  *
+ *   - reconfig: (optional) called if requested to reload configuration
  */
 struct exec_implementation {
     const char *name;
-    int  (*config)  (flux_t *h, int argc, char **argv);
-    void (*unload)  (void);
-    int  (*init)    (struct jobinfo *job);
-    void (*exit)    (struct jobinfo *job);
-    int  (*start)   (struct jobinfo *job);
-    int  (*kill)    (struct jobinfo *job, int signum);
-    int  (*cancel)  (struct jobinfo *job);
-    int  (*stats)   (json_t **stats);
+    int  (*config)   (flux_t *h, int argc, char **argv);
+    void (*unload)   (void);
+    int  (*init)     (struct jobinfo *job);
+    void (*exit)     (struct jobinfo *job);
+    int  (*start)    (struct jobinfo *job);
+    int  (*kill)     (struct jobinfo *job, int signum);
+    int  (*cancel)   (struct jobinfo *job);
+    int  (*stats)    (json_t **stats);
+    int  (*reconfig) (flux_t *h,
+                      const flux_conf_t *conf,
+                      int argc,
+                      char **argv,
+                      flux_error_t *errp);
```

Comment:
Oh, if you add the config to the handle as mentioned in another commit, then maybe we don't need a different `config` and `reconfig` operations?